apps/server/src/modules/deletion/repo/deletion-log.repo.spec.ts

```diff
@@ -0,0 +1,186 @@
+					modifiedCount: entity.modifiedCount,
+					deletedCount: entity.deletedCount,
+					deletionRequestId: entity.deletionRequestId?.toHexString(),
+					createdAt: entity.createdAt,
+					updatedAt: entity.updatedAt,
+				};
+
+				return {
+					entity,
+					expectedDeletionLog,
+				};
+			};
+
+			it('should find the deletionRequest', async () => {
+				const { entity, expectedDeletionLog } = await setup();
+
+				const result: DeletionLog = await repo.findById(entity.id);
+
+				// Verify explicit fields.
+				expect(result).toEqual(expect.objectContaining(expectedDeletionLog));
+			});
+		});
+	});
+
+	describe('findAllByDeletionRequestId', () => {
+		describe('when there is no deletionLog for deletionRequestId', () => {
+			it('should return empty array', async () => {
+				const deletionRequestId = new ObjectId().toHexString();
+				const result = await repo.findAllByDeletionRequestId(deletionRequestId);
+
+				expect(result).toEqual([]);
+			});
+		});
+
+		describe('when searching by deletionRequestId', () => {
+			const setup = async () => {
+				const deletionRequest1Id = new ObjectId();
+				const deletionRequest2Id = new ObjectId();
+				const deletionLogEntity1: DeletionLogEntity = deletionLogEntityFactory.build({
+					deletionRequestId: deletionRequest1Id,
+				});
+				const deletionLogEntity2: DeletionLogEntity = deletionLogEntityFactory.build({
+					deletionRequestId: deletionRequest1Id,
+				});
+				const deletionLogEntity3: DeletionLogEntity = deletionLogEntityFactory.build({
+					deletionRequestId: deletionRequest2Id,
+				});
+
+				await em.persistAndFlush([deletionLogEntity1, deletionLogEntity2, deletionLogEntity3]);
+				em.clear();
+
+				return { deletionLogEntity1, deletionLogEntity2, deletionLogEntity3, deletionRequest1Id };
+			};
+
+			it('should find deletionRequests with deleteAfter smaller then today', async () => {
+				const { deletionLogEntity1, deletionLogEntity2, deletionLogEntity3, deletionRequest1Id } = await setup();
+
+				const results = await repo.findAllByDeletionRequestId(deletionRequest1Id.toHexString());
+
+				const expectedArray = [
+					{
+						id: deletionLogEntity1.id,
+						domain: deletionLogEntity1.domain,
+						operation: deletionLogEntity1.operation,
+						deletionRequestId: deletionLogEntity1.deletionRequestId?.toHexString(),
+						modifiedCount: deletionLogEntity1.modifiedCount,
+						deletedCount: deletionLogEntity1.deletedCount,
+						createdAt: deletionLogEntity1.createdAt,
+						updatedAt: deletionLogEntity1.updatedAt,
+					},
+					{
+						id: deletionLogEntity2.id,
+						domain: deletionLogEntity2.domain,
+						operation: deletionLogEntity2.operation,
+						deletionRequestId: deletionLogEntity2.deletionRequestId?.toHexString(),
+						modifiedCount: deletionLogEntity2.modifiedCount,
+						deletedCount: deletionLogEntity2.deletedCount,
+						createdAt: deletionLogEntity2.createdAt,
+						updatedAt: deletionLogEntity2.updatedAt,
+					},
+				];
```

Comment:
Please move this to the `setup()` function.

Reply:
done